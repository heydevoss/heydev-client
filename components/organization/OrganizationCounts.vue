<template>
  <div class="organization-counts">
    <div v-for="count in counts" :key="count.label" class="org-count">
      <div class="org-count--item">
        <span class="value">
          {{ count.value }}
        </span>
        <span class="label">
          {{ count.label }}
        </span>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'OrganizationCounts',
  props: {
    totalPullRequests: Number,
    totalCommits: Number,
    totalIssues: Number,
  },
  data() {
    return {
      counts: [
        { label: 'Issues', value: this.totalIssues },
        { label: 'Pull Requests', value: this.totalPullRequests },
        { label: 'Commits', value: this.totalCommits },
      ]
    }
  }
}
</script>

<style lang="scss">
.organization-counts {
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
}
.org-count {
  display: flex;
  flex-direction: column;
}
.org-count--item {
  display: flex;
  flex-direction: column;
}

.value {
  font-size: calc(2.0em + 1vw);
  color: #FFA600;
}

.label {
  margin-top: 4px;
  font-size: calc(0.2em + 1vw);
}

@media screen and (max-width: 820px) {
  .organization-counts {
    flex-direction: column;
    justify-content: space-between;
  }  
  .label {
    font-size: calc(1em + 1vw);
  }
}
</style>
