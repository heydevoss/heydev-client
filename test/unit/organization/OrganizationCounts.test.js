import { shallowMount } from '@vue/test-utils'
import OrganizationCounts from '@/components/organization/OrganizationCounts.vue'

describe('<OrganizationCounts/>', () => {
  const wrapper = shallowMount(OrganizationCounts, {
    propsData: {
      totalPullRequests: 13,
      totalCommits: 2,
      totalIssues: 4
    }
  })

  it('should be a Vue instance and renders correctly', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render counts', () => {
    expect(wrapper.html()).toContain('13')
    expect(wrapper.html()).toContain('2')
    expect(wrapper.html()).toContain('4')
  })

  it('should render labels', () => {
    expect(wrapper.html()).toContain('Issues')
    expect(wrapper.html()).toContain('Pull Requests')
    expect(wrapper.html()).toContain('Commits')
  })

  it('should set values to 0 when its not passed the counts', () => {
    const wrapperNew = shallowMount(OrganizationCounts)
    expect(wrapperNew.html().includes('span class="value">0</span>'))
    expect(wrapper.html()).toContain('Issues')
    expect(wrapperNew.html()).toContain('Pull Requests')
    expect(wrapperNew.html()).toContain('Commits')
  })
})
