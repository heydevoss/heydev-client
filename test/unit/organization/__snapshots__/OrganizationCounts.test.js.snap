// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrganizationCounts/> should be a Vue instance and renders correctly 1`] = `
<div
  class="organization-counts"
>
  <div
    class="org-count"
  >
    <div
      class="org-count--item"
    >
      <span
        class="value"
      >
        
        4
      
      </span>
       
      <span
        class="label"
      >
        
        Issues
      
      </span>
    </div>
  </div>
  <div
    class="org-count"
  >
    <div
      class="org-count--item"
    >
      <span
        class="value"
      >
        
        13
      
      </span>
       
      <span
        class="label"
      >
        
        Pull Requests
      
      </span>
    </div>
  </div>
  <div
    class="org-count"
  >
    <div
      class="org-count--item"
    >
      <span
        class="value"
      >
        
        2
      
      </span>
       
      <span
        class="label"
      >
        
        Commits
      
      </span>
    </div>
  </div>
</div>
`;
